package/tests/test_server.py

```diff
@@ -188,3 +189,52 @@ def test_nodes_endpoint(client):
     assert response.status_code == 200
     data = json.loads(response.data.decode())
     assert data == EXPECTED_PIPELINE_DATA
+
+
+@pytest.fixture(autouse=True)
+def clean_up():
+    server.VIZ_THREADS = {}
+
+
+def test_wait_for():
+    def _sum(x, y):
+        return x + y
+
+    assert not server.wait_for(_sum, 3, x=1, y=2)
+
+    unexpected_result_error = r"didn\'t return 0 within specified timeout"
+    with pytest.raises(WaitForException, match=unexpected_result_error):
+        server.wait_for(_sum, 0, x=1, y=2, timeout_=1)
+
+    # Non-callable should fail
+    non_callable = 1
+    non_callable_error = r"didn\'t return True within specified timeout"
+    with pytest.raises(WaitForException, match=non_callable_error):
+        server.wait_for(non_callable, timeout_=1)
+
+
+def test_run_viz(mocker):
+    """Test inline magic function"""
+    mocked_thread = mocker.patch(
+        "kedro_viz.server.threading.Thread", return_value=mocker.Mock()
+    )
+    _ = mocker.patch("kedro_viz.server.wait_for")
+    default_port = 4141
+    server.run_viz()
+
+    # pylint: disable=protected-access
+    mocked_thread.assert_called_with(
+        target=server._call_viz, kwargs={"port": default_port}, daemon=True
+    )
+
+    # Running run_viz with the same port should not trigger another thread
+    server.run_viz()
+    assert mocked_thread.call_count == 1
```

Comment:
If you do `.reset_mock()`, then this assertion would make sense: `mocked_thread.assert_not_called()`